### PR TITLE
Use GOCACHE in .circleci

### PR DESCRIPTION
**Pull request by krhubert**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,13 +13,23 @@ runOnDocker: &runOnDocker
...
       - run: docker swarm init
       - run: docker build -t sleep docker-images/sleep/
       - run: docker build -t http-server docker-images/http-server/
+      - restore_cache:
+          keys:
+            - build-cache-{{ .Branch  }}-{{ .Environment.CIRCLE_PREVIOUS_BUILD_NUM }}
+            - test-cache-{{ .Branch }}
       - run: go test -v -timeout 300s -p 1 -tags=integration -coverprofile=coverage.txt ./...
+      - save_cache:
+          key: test-cache-{{ .Branch }}-{{ .BuildNum }}
```
[the remark this answers is not in the document]
Comment (krhubert) [reply .circleci/config.yml]: > does this actually works ?

What do you mean? You can read about GOCACHE at go offical pages.

> so is it really possible to restore the cache from test-cache-{{ .Branch }} ?

Yes, https://circleci.com/docs/2.0/caching/#restoring-cache -> **Each cache key is namespaced to the project, and retrieval is prefix-matched.**

> The `build-cache-....`

Changed to `test-cache`

[the remark this answers is not in the document]
Comment (antho1404) [reply .circleci/config.yml]: yes my point was really about the modification you did 👍

**Comment by krhubert** [comment]
I have removed coverprofile because we don't use it now and go test cache doesn't work with it.

It has open issues about it:
- https://github.com/golang/go/issues/23565
- https://go-review.googlesource.com/c/go/+/90955/

So we need to remove coverprofile to speed up the tests or wait for go 1.12 (or maybe even 1.13, because I don't see this issue on 1.12 milestone)

**Comment by krhubert** [comment]
As decided, we keep coverprofile and waits to go team to fix the issues.

**Pull request closed without merging**